engine/lib/user_settings.php

```diff
@@ -85,6 +85,10 @@ function _elgg_set_user_name() {
 	$name = strip_tags(get_input('name'));
 	$user_guid = get_input('guid');
 
+	if (!isset($name)) {
```

Comment:
I think you'll need to check this before `strip_tags` (I assume will output empty string).